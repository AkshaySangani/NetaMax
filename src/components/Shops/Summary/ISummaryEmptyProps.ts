export interface ISummaryEmptyProps {
  /**
   * The Url Link
   * @type {String}
   * */
  shopUrl: string;
  /**
   * The shopName
   * @type {String}
   * */
  shopName: string | undefined;
  /**
   * The shortStoreName
   * @type {String}
   * */
  shortStoreName: string | undefined;
  /**
   * The hasSummary
   * @type {Boolean}
   * */
  hasSummary: boolean | undefined;

  /**
   * The add event to share ligue
   * @type {() => void}
   **/
  onClickCompartirLiga: () => void;
}
