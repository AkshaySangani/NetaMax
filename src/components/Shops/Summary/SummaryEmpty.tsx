import React, { ReactElement } from "react";
import IconTracking from "styled/icons/Shops/Summary/Tracking";
import { Box, Button, Flex, Link, Text } from "@chakra-ui/react";
import { ISummaryEmptyProps } from "./ISummaryEmptyProps";

/**
 * The SummaryEmpty shops component.
 * @param {INavBarShopsProps} props The component props
 * @returns {React.ReactElement} The SummaryEmpty shops component
 */
export const SummaryEmpty = (props: ISummaryEmptyProps): ReactElement => {
  const { shopUrl, shopName, hasSummary, onClickCompartirLiga } = props;
  const shortStoreName = shopName?.split(".")[0];

  return hasSummary ? (
    <Flex justifyContent="space-around" direction={["column"]} align={["center", "flex-start"]}>
      <Box
        boxShadow="base"
        p={["2", "3"]}
        rounded="md"
        bg="white"
        w={["100%", "405px"]}
        minW="343px"
        mt="6px"
      >
        <Flex direction={["column"]} w="100%">
          <Flex direction="column">
            <Text fontWeight="bold">Aún no tienes ventas para este día</Text>
            <Text>
              Invita a tus clientes a comprar en tu liga para aumentar las ganancias de tu tiendita.
            </Text>
          </Flex>
        </Flex>
      </Box>
      <Flex
        justifyContent="space-around"
        mr="4px"
        mt="34px"
        mb="14px"
        bg="#F7F6F8"
        borderRadius="24px"
        w={["95%", "343px"]}
        h="26px"
        minW="343px"
      >
        wwww.neta.mx/{shortStoreName}
      </Flex>
      <Link href={shopUrl} passHref textDecoration="none!important">
        <Button
          color="#3870ff"
          border="1px solid #3870FF"
          _focus={{ border: "1px solid #3870FF" }}
          bg="#3870ff"
          mr={["", "2"]}
          p="3"
          mt={["2", "3"]}
          mb={["2", "1"]}
          borderRadius="16px"
          h={["40px", "56px"]}
          w="343px"
          minW="343px"
          onClick={() => onClickCompartirLiga()}
        >
          <IconTracking />
          <Text color="white">Comparte tu liga</Text>
        </Button>
      </Link>
    </Flex>
  ) : (
    <Box boxShadow="base" p={["2", "3"]} rounded="md" bg="white" w={["100%", "405px"]} mt="6px">
      <Flex direction={["column", "row", "row"]}>
        <Flex direction="column">
          <Text fontWeight="bold">Las ventas para esta fecha se cerraron</Text>
          <Text>
            Ninguna venta se registró en tu tiendita para este día, selecciona otra fecha para ver
            tus ventas.
          </Text>
        </Flex>
      </Flex>
    </Box>
  );
};
