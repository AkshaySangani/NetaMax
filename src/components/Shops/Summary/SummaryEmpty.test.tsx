import { render, cleanup, screen } from "@testing-library/react";
import { SummaryEmpty } from "./SummaryEmpty";
import { ISummaryEmptyProps } from "./ISummaryEmptyProps";

describe("SummaryEmpty Component", () => {
  let mock: ISummaryEmptyProps;

  beforeEach(() => {
    mock = {
      shopUrl: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com",
      shopName: "Shop.One",
      shortStoreName: "SO",
      hasSummary: true,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("SummaryEmpty should be render correctly with hasSummary flag false", () => {
    mock = { ...mock, hasSummary: false };
    const { container } = render(<SummaryEmpty {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("SummaryEmpty should be render correctly with hasSummary flag", () => {
    const { container } = render(<SummaryEmpty {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("ShortShopName should be present in document", () => {
    render(<SummaryEmpty {...mock} />);
    expect(screen.getByText(`wwww.neta.mx/${mock.shopName?.split(".")[0]}`)).toBeInTheDocument();
  });

  test("Comparte tu liga Link should be render properly", () => {
    render(<SummaryEmpty {...mock} />);
    expect(screen.getByRole("link")).toHaveAttribute("href", mock.shopUrl);
  });
});
