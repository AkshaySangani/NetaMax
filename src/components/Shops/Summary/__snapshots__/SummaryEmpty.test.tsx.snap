// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryEmpty Component SummaryEmpty should be render correctly with hasSummary flag 1`] = `
<div>
  <div
    class="css-12h7owt"
  >
    <div
      class="css-nt4f9c"
    >
      <div
        class="css-1ofqig9"
      >
        <div
          class="css-j7qwjs"
        >
          <p
            class="chakra-text css-in3yi3"
          >
            Aún no tienes ventas para este día
          </p>
          <p
            class="chakra-text css-0"
          >
            Invita a tus clientes a comprar en tu liga para aumentar las ganancias de tu tiendita.
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-ks45v5"
    >
      wwww.neta.mx/
      Shop
    </div>
    <a
      class="chakra-link css-fx7esi"
      href="https://nopcommerce-assets.s3.us-east-2.amazonaws.com"
    >
      <button
        class="chakra-button css-1y2lowl"
        type="button"
      >
        <svg
          fill="none"
          height="23"
          viewBox="0 0 20 23"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.1289 16.0449C15.0864 17.0874 12.875 19.2988 11.4135 20.7603C10.6325 21.5414 9.36862 21.5412 8.58757 20.7601C7.15127 19.3238 4.9841 17.1566 3.87239 16.0449C0.487849 12.6604 0.487849 7.17295 3.87239 3.78841C7.25694 0.403864 12.7444 0.403864 16.1289 3.78841C19.5135 7.17295 19.5135 12.6604 16.1289 16.0449Z"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
          <path
            d="M13.2507 9.91667C13.2507 11.7116 11.7956 13.1667 10.0007 13.1667C8.20573 13.1667 6.75065 11.7116 6.75065 9.91667C6.75065 8.12174 8.20573 6.66667 10.0007 6.66667C11.7956 6.66667 13.2507 8.12174 13.2507 9.91667Z"
            stroke="white"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <p
          class="chakra-text css-kbn7if"
        >
          Comparte tu liga
        </p>
      </button>
    </a>
  </div>
</div>
`;

exports[`SummaryEmpty Component SummaryEmpty should be render correctly with hasSummary flag false 1`] = `
<div>
  <div
    class="css-1y150zw"
  >
    <div
      class="css-sm9syx"
    >
      <div
        class="css-j7qwjs"
      >
        <p
          class="chakra-text css-in3yi3"
        >
          Las ventas para esta fecha se cerraron
        </p>
        <p
          class="chakra-text css-0"
        >
          Ninguna venta se registró en tu tiendita para este día, selecciona otra fecha para ver tus ventas.
        </p>
      </div>
    </div>
  </div>
</div>
`;
